pkg/controller/tas/node_failure_controller.go

```diff
@@ -187,20 +191,28 @@ func (r *nodeFailureReconciler) patchWorkloadsForUnavailableNode(ctx context.Con
 			}
 			continue
 		}
-
+		annotations := wl.GetAnnotations()
+		if annotations == nil {
+			annotations = make(map[string]string)
+		}
+		failedNode, ok := annotations[kueuealpha.NodeToReplaceAnnotation]
+		if ok && failedNode != nodeName && !workload.IsEvicted(&wl) {
```

Comment:
please reduce scope of these variables, and fetch again below (inside the function provided to patch). This is especially important for readability of !ok below.

Or other option is to rename the ok back to something more verbose. I prefer more local variable option, however - it makes the 2nd block easier to read as you don't need to scroll up to see what the variables are.
```suggestion
		if failedNode, ok := annotations[kueuealpha.NodeToReplaceAnnotation]; ok && failedNode != nodeName && !workload.IsEvicted(&wl) {
```

Reply:
Done (first option).